### DISCUSSION TITLE
Welcome to Koin project Discussions!

### DISCUSSION THREAD

**Discussion by arnaudgiuliani** [InsertKoinIO/koin#1013] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with all members of our community. We hope that you:
  * Team announcement & feature topics 
  * Share ideas & feedback to help improve
  * Ask questions you’re wondering about
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪

Found a bug or a problem on a specific feature? [Open an issue on Github issues](https://github.com/InsertKoinIO/koin/issues/new/choose)

**Comment by desugar-64** [comment]
Koin captivates with its simplicity, clean API and low learning curve. I like Koin and have been using it for about a year. 
Also, I have experience working with Dagger and Hilt. They are good for what they were made for, but they greatly slow down the compilation time of projects because they use code generation in the form of an annotation processor, which works even slower with Kotlin in the guise of Kapt.

What do you think about to provide the desired compile-time safety by moving Koin onto the [KSP ](https://github.com/google/ksp)rails?

Thanks.

**Reply by arnaudgiuliani** [reply]
It's the next priority for Koin to be powered by Kotlin compiler plugin and perhaps some annotations.
